addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -450,6 +450,22 @@ var SnippetEditor = Widget.extend({
 
         return Promise.all(defs);
     },
+    /**
+     * Returns true if the element matches a snippet block that cannot be
+     * droppped in a sanitized HTML field.
+     *
+     * @param {jQuery} $el
+     * @return {boolean}
+     */
+    _isForbidSanitize($el) {
+        const $scroll = $('#o_scroll');
+        for (const cl of $el[0].classList) {
+            if ($scroll.find('.oe_snippet_body.' + cl).closest('.oe_snippet').find('.oe_forbid_sanitize').length) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
Don't search the whole DOM to find editor information. The responsibility of knowing information about snippets is in the `SnippetMenu` class. If you need one of those info in the `SnippetEditor` class, then do a `trigger_up`, handled by the `SnippetMenu`.